app/src/main/java/com/davidtomas/taskyapp/core/presentation/util/DateUtils.kt

```diff
@@ -17,6 +20,40 @@ fun Long?.formatToMMDDYY(): String {
     return defaultZoneDateTime.format(DateTimeFormatter.ofPattern("MMM dd yyyy"))
 }
 
+fun ZonedDateTime.formatToMMM(): String {
+    return this.format(DateTimeFormatter.ofPattern("MMMM")).uppercase(Locale.ROOT)
+}
+
+fun ZonedDateTime.daysOfWeekIncludingGivenDate(): Map<DayOfWeek, ZonedDateTime> {
+    val days = mutableListOf<ZonedDateTime>()
+    val startOfWeek = this.with(TemporalAdjusters.previousOrSame(DayOfWeek.MONDAY))
+    val endOfWeek = this.with(TemporalAdjusters.nextOrSame(DayOfWeek.SUNDAY))
+    var currentDate = startOfWeek
+    while (!currentDate.isAfter(endOfWeek)) {
+        days.add(currentDate)
+        currentDate = currentDate.plusDays(1)
+    }
+    return DayOfWeek.values().zip(days).toMap()
+}
```

Comment:
What speaks against just having a `List<ZonedDateTime>` for the items that are displayed in the day picker row? Why do you need this more complex logic to parse it into a map?